crates/core/prelude/text-encoding.js

```diff
@@ -47,8 +47,9 @@
             });
         }
 
-        encode(input) {
+        encode(input = "") {
             input = input.toString(); // non-string inputs are converted to strings
+            input = replaceUnpairedSurrogates(input); // QuickJS does not handle this correctly
```

Comment:
Sorry to question this, but are you sure about this? QuickJS passes the ECMA262 test suite, so I'd be surprised if there was any non-standard behavior in QuickJS. 

Isn't this exactly what WTF-16 allows, i.e. lone surrogates?